frontend/utility/MissingFilesModel.cpp

```diff
@@ -214,32 +236,32 @@ bool MissingFilesModel::setData(const QModelIndex &index, const QVariant &value,
 		endInsertRows();
 
 		success = true;
-	} else {
-		QString path = value.toString();
-		if (index.column() == MissingFilesColumn::NewPath) {
-			files[index.row()].newPath = value.toString();
-			QString fileName = QUrl(path).fileName();
-			QString origFileName = QUrl(files[index.row()].originalPath).fileName();
-
-			if (path.isEmpty()) {
-				files[index.row()].state = MissingFilesState::Missing;
-			} else if (path.compare(QTStr("MissingFiles.Clear")) == 0) {
-				files[index.row()].state = MissingFilesState::Cleared;
-			} else if (fileName.compare(origFileName) == 0) {
-				files[index.row()].state = MissingFilesState::Found;
-
-				if (loop)
-					fileCheckLoop(files, path, false);
-			} else {
-				files[index.row()].state = MissingFilesState::Replaced;
-
-				if (loop)
-					fileCheckLoop(files, path, false);
-			}
+	} else if (index.column() == MissingFilesColumn::NewPath && files[index.row()].newPath != value.toString()) {
 
-			emit dataChanged(index, index);
-			success = true;
+		QString path = value.toString();
+		files[index.row()].newPath = value.toString();
+		QString fileName = QUrl(path).fileName();
+		QString origFileName = QUrl(files[index.row()].originalPath).fileName();
+
+		if (path.isEmpty()) {
+			files[index.row()].state = MissingFilesState::Missing;
+		} else if (path.compare(QTStr("MissingFiles.Clear")) == 0) {
+			files[index.row()].state = MissingFilesState::Cleared;
+		} else if (fileName.compare(origFileName) == 0) {
+			files[index.row()].state = MissingFilesState::Found;
+
+			if (fileLoopCounter == 0)
+				fileCheckLoop(files, path, false);
```

Comment:
I'd prefer if even single-line conditional blocks are wrapped in curly braces. There is no good reason to omit them these days, so much so that modern languages don't allow that practice at all anymore for good reason.

Maintenance-wise it is the smart decision to _always_ use curly braces to make flow control explicit.

Reply:
It's honestly my own preference as well